### DISCUSSION TITLE
How to consume pending messages in a queue after the consumer reconnects to the broker?

### DISCUSSION THREAD

**Discussion by f-uba** [rabbitmq/rabbitmq-dotnet-client#1279] [Q&A]
I'm testing some features in my software and I'm using the consumer (EventingBasicConsumer) events to get new messages from a queue, but if my consumer lost connection and after a while reconnect, it don't get the pending message on the queue (with it have one), the payloads stays as unacked and I can't do nothing with it. There's any suggestions on how to handle that situation?

I used the BasicRecover method from IModel class and it works, but the comments of the method itself say not to use it, so it is not a valid solution. _"Indicates that a consumer has recovered. Deprecated. Should not be used."_

**Comment by lukebakken** [comment]
How are you reconnecting? Can you share your code?

When a connection is lost RabbitMQ _will_ re-queue the message and put it back into the Ready state. My guess is there is a bug in your application.

What versions are you using of...
* RabbitMQ
* Erlang
* .NET client
* Operating system

**Comment by michaelklishin** [comment]
See [Automatic Re-queueing](https://www.rabbitmq.com/confirms.html#automatic-requeueing)
